engine.py

```diff
@@ -891,3 +930,128 @@ def _get_dialog_parent(self):
             "Found top level widget %s for dialog parenting" % (parent,))
         return parent
 
+    def node_handler(self, node):
+        """
+        Get the node handler hook for the given node. 
+
+        :param node: A :class:`hou.Node` instance.
+
+        :returns: A :class:`NodeHandlerBase` instance.
+        """
+        if not node: 
+            return
+        node_type = node.type()
+        node_type_name = node_type.name()
+        node_category = node_type.category().name()
+        hook_instance = self.__node_handlers.get(node_category, {}).get(node_type_name, False)
+
+        if hook_instance is False:
+            if not self.__node_handlers.get(node_category):
+                self.__node_handlers[node_category] = {}
+            node_handlers = self.get_setting("node_handlers")
+            hook_instance = None
+            for handler in node_handlers:
+                setup_hook = (
+                    handler["node_category"] == node_category
+                    and handler["node_type"] == node_type_name
+                )
+                if setup_hook:                
+                    tk_houdini = self.import_module("tk_houdini")
+                    base_class = tk_houdini.base_hooks.NodeHandlerBase
+                    hook_instance = self.create_hook_instance(
+                        handler["hook"],
+                        base_class=base_class,
+                    )
+                    break
+            self.__node_handlers[node_category][node_type_name] = hook_instance
+                    
+        return hook_instance
+    
+    def all_sgtk_nodes(self):
+        """
+        Get all teh nodes in the scene that contains sgtk parameters.
+
+        :rtype: list(:class:`hou.Node`)
+        """
+        node_type_categories = hou.nodeTypeCategories()
+        node_handler_settings = self.get_setting("node_handlers")
+
+        def key_func(item):
+            return item["node_category"]
+
+        grouped = itertools.groupby(
+            sorted(node_handler_settings, key=key_func),
+            key=key_func
+        )
+
+        for category_name, node_handlers in grouped:
+            category = node_type_categories[category_name]
+            for node_handler in node_handlers:
+                node_type_name = node_handler["node_type"]
+                node_type = hou.nodeType(category, node_type_name)
+                for node in node_type.instances():
+                    parm = node.parm("sgtk_identifier")
+                    if parm:
+                        self.log_debug(node.path())
```

Comment:
`self.log_debug` [is deprecated](https://developer.shotgunsoftware.com/tk-core/platform.html#sgtk.platform.Engine.log_debug)
```suggestion
                        self.logger.debug(node.path())
```